lib/Elastica/Suggest/CandidateGenerator/DirectGenerator.php

```diff
@@ -89,7 +89,7 @@ public function setMinWordLength($min)
      *
      * @return $this
      */
-    public function setMaxInspections($max)
+    public function setMaxInspections(int $max)
```

Comment:
- max_inspections
- min_doc_freq
- max_term_freq
have their own default value

Reply:
didn't get this point. we should make it like `setMaxInspections(int $max = null)` and check field and depends on field set default value?

Reply:
https://github.com/ruflin/Elastica/pull/1579#discussion_r243801054
@osavchenko checkout this discussion about the default values :)

Reply:
So, I need to define all default values as constants like `const DEFAULT_MAX_INSPECTIONS = some_value`?

Reply:
yep that's the idea

Reply:
Exactly. This for every value required by @p365labs